proc-macros/src/render_client.rs

```diff
@@ -68,6 +68,36 @@ impl RpcDescription {
 		Ok(trait_impl)
 	}
 
+	fn patch_result_error(&self, ty: &syn::Type) -> syn::Type {
+		let mut path = match ty {
+			syn::Type::Path(path) => path.clone(),
+			_ => panic!("Client only supports bare or Result values: {:?}", ty),
+		};
+
+		let Some(first_segment) = path.path.segments.first_mut() else {
+			return syn::Type::Path(path);
+		};
+
+		if first_segment.ident != "Result" {
```

Comment:
I'm wondering would this work as well for `std::result::Result`?

Reply:
This is a very good find! In fact, this logic has to be altered to permit a number of paths here, including `std::result::Result` - however I need more research into proc macros to determine exactly what things to cover, so I left it for now; this is one of the TODOs before the merge.

Reply:
This was addressed by using the last argument